core/lib/dal/src/applied_snapshot_status_dal.rs

```diff
@@ -0,0 +1,100 @@
+use zksync_types::{snapshots::AppliedSnapshotStatus, L1BatchNumber};
+
+use crate::StorageProcessor;
+
+#[derive(Debug)]
+pub struct AppliedSnapshotStatusDal<'a, 'c> {
+    pub(crate) storage: &'a mut StorageProcessor<'c>,
+}
+
+impl AppliedSnapshotStatusDal<'_, '_> {
+    pub async fn set_applied_snapshot_status(
+        &mut self,
+        status: &AppliedSnapshotStatus,
+    ) -> Result<(), sqlx::Error> {
+        sqlx::query!(
+            r#"
+            INSERT INTO applied_snapshot_status(l1_batch_number, is_finished, last_finished_chunk_id)
+            VALUES ($1, $2, $3)
+            ON CONFLICT
+                (l1_batch_number)
+                DO UPDATE
+                SET l1_batch_number        = excluded.l1_batch_number,
+                    is_finished            = excluded.is_finished,
+                    last_finished_chunk_id = excluded.last_finished_chunk_id
+            "#,
+            status.l1_batch_number.0 as i64,
+            status.is_finished,
+            status.last_finished_chunk_id.map(|v| v as i32)
+        )
+            .execute(self.storage.conn())
+            .await
+            .unwrap();
```

Comment:
- Nit: Indentation should be 4 spaces smaller here.
- Please propagate an error instead of `unwrap()`ping it.

Reply:
corrected